hi-ml-azure/src/health_azure/utils.py

```diff
@@ -1252,3 +1255,51 @@ def torch_barrier() -> None:
         return
     if distributed.is_available() and distributed.is_initialized():
         distributed.barrier()
+
+
+def aggregate_hyperdrive_metrics(run_id: str, workspace: Workspace) -> pd.DataFrame:
+    """
+    For a given HyperDrive run id, retrieves the metrics from each of its children and then aggregates it.
+    Returns a DataFrame where each column is one child run, and each row is a metric logged by that child run.
+    For example, for a HyperDrive run with 2 children, where each logs epoch, accuracy and loss, the result
+    would look like:
+
+    |              | 0               | 1                  |
+    |--------------|-----------------|--------------------|
+    | epoch        | [1, 2, 3]       | [1, 2, 3]          |
+    | accuracy     | [0.7, 0.8, 0.9] | [0.71, 0.82, 0.91] |
+    | loss         | [0.5, 0.4, 0.3] | [0.45, 0.37, 0.29] |
```

Comment:
your examples assume that all metrics have the same length. What happens if that is not the case?

Reply:
Maybe it would be better to return a dictionary mapping from metric name to dataframe?

Reply:
(discussed offline)